server/src/main/java/org/elasticsearch/cluster/metadata/MetadataDataStreamsService.java

```diff
@@ -91,6 +142,38 @@ static ClusterState modifyDataStream(
         return ClusterState.builder(currentState).metadata(updatedMetadata).build();
     }
 
+    /**
+     * Creates an updated cluster state in which the requested data streams have the data lifecycle provided.
+     * Visible for testing.
+     */
+    static ClusterState modifyDataLifecycle(
+        ClusterState currentState,
+        List<String> dataStreamNames,
+        @Nullable DataLifecycle dataLifecycle
+    ) {
+        Metadata updatedMetadata = currentState.metadata();
+        Metadata.Builder builder = Metadata.builder(updatedMetadata);
+        for (var dataStreamName : dataStreamNames) {
+            var dataStream = validateDataStream(updatedMetadata, dataStreamName);
```

Comment:
This throws an exception if a data stream in the list is missing. 
Shall we document this for all APIs? (we currently say that data streams without a lifecycle will not be included in the response - shall we document that requesting a missing data stream in the list will yield 404?

Reply:
Yes, I added it.